build.xml

```diff
@@ -888,8 +888,7 @@
     <!--
         The build target builds all the .class files
     -->
-    <target name="build" depends="resolver-retrieve-build,build-project,checkstyle" description="Compile Cassandra classes"/>
-    <target name="_build_unsafe" depends="resolver-retrieve-build,build-project" description="Compile Cassandra classes without checks"/>
+    <target name="build" depends="resolver-retrieve-build,build-project" description="Compile Cassandra classes"/>
```

Comment:
a concern i have is that by removing `checkstyle` from `build`, `rat-check` from `build-project`, and `eclipse-warnings` from `jar`, we are less likely to be catching these failures before commits are pushed. because folk are not often running `artifacts` locally. even if CI catches it, it is still nice to have a faster dev feedback loop on such failures. also worth noting that circleci does not run artifacts.

so the `_build_unsafe` and `*.skip` approach has appeal to me. leaning me towards instead adding the appropriate skip flags to all the ant calls in https://github.com/apache/cassandra-builds/blob/trunk/build-scripts/cassandra-test.sh and https://github.com/apache/cassandra-builds/blob/trunk/build-scripts/cassandra-dtest-pytest.sh

Reply:
ok, so I think that the default `jar` may depend on all checks, but `build`, `build-test`, `_main_jar` should not. This way we can avoid the need to add extra properties while keeping the default task, and CircleCI run the checks without the need to call `artifacts` goal.